### PR TITLE
[Workbase] - Initialise GRPC transactions locally and attach attributes to nodes

### PR DESCRIPTION
**Pull request by Irtazaraza**
# Why is this PR needed?
A GRPC transaction was only being opened once and hence Workbase did not register any real-time changes to the graph

# What does the PR do?
- Open and close transactions locally for whenever API calls are needed to be made
- Attach attributes to nodes when they are queried initially so attributes of inferred types are persisted 

# Does it break backwards compatibility?
- No

# List of future improvements not on this PR
- No

### CODE REVIEW

File: workbase/src/renderer/components/Visualiser/RightBar/SettingsTab/DisplaySettingsPanel.vue [hunk 1]
```diff
@@ -126,11 +126,15 @@
     },
     methods: {
       async loadAttributeTypes() {
+        await this.localStore.openGraknTx();
```
Comment (marco-scoppetta) [inline workbase/src/renderer/components/Visualiser/RightBar/SettingsTab/DisplaySettingsPanel.vue]: local tx

File: workbase/src/renderer/components/Visualiser/VisualiserGraphBuilder.js [hunk 2]
```diff
@@ -72,21 +84,28 @@ async function buildLabel(node) {
...
   rel.type = await (await rel.type()).label();
+  rel.attributes = await computeAttributes(rel);
+  rel.isInferred = await rel.isInferred();
 }
 
 async function prepareAttribute(attribute) {
   attribute.type = await (await attribute.type()).label();
   attribute.value = await attribute.value();
   attribute.label = await buildLabel(attribute);
+  attribute.attributes = await computeAttributes(attribute);
```
Comment (marco-scoppetta) [inline workbase/src/renderer/components/Visualiser/VisualiserGraphBuilder.js]: blocking everything waiting on all the attributes to be loaded might become really heavy, specially with high latency network - it's fine for now but need to implement a lazy loading in background that doesnt block ASAP

File: workbase/src/renderer/components/Visualiser/VisualiserStore.js [hunk 3]
```diff
@@ -109,17 +154,29 @@ const methods = {
...
+  },
+  async closeGraknTx() {
+    await this.graknTx.close();
+  },
 
   // getters
+  getMetaTypeInstances() {
+    return this.metaTypeInstances;
+  },
+  async getNode(nodeId) {
```
Comment (marco-scoppetta) [inline workbase/src/renderer/components/Visualiser/VisualiserStore.js]: async without await, no need

File: workbase/src/renderer/components/Visualiser/VisualiserStore.js [hunk 4]
```diff
@@ -146,6 +203,8 @@ const state = {
   currentQuery: '',
   loadingQuery: false,
   explanationQuery: false,
+  graknTx: undefined,
```
Comment (marco-scoppetta) [inline workbase/src/renderer/components/Visualiser/VisualiserStore.js]: let's hold a reference to session, not tx no more

File: workbase/src/renderer/components/shared/CanvasStoreMixin/CanvasStoreMixin.js [hunk 5]
```diff
@@ -17,26 +21,10 @@ const actions = {
       if (keyspace) { // keyspace will be null if user deletes current keyspace from Keyspaces page
         this.graknSession = GlobalStore.state.grakn.session(this.currentKeyspace);
         this.openGraknTx();
+        await this.dispatch(LOAD_METATYPE_INSTANCES);
```
Comment (marco-scoppetta) [inline workbase/src/renderer/components/shared/CanvasStoreMixin/CanvasStoreMixin.js]: no need to await here

File: workbase/src/renderer/components/Visualiser/VisualiserStore.js [hunk 6]
```diff
@@ -1,3 +1,4 @@
+import Grakn from 'grakn';
```
[the remark this answers is not in the document]
Comment (Irtazaraza) [reply workbase/src/renderer/components/Visualiser/VisualiserStore.js]: "Not a big deal" :p

File: workbase/src/renderer/components/Visualiser/RightBar/SettingsTab/DisplaySettingsPanel.vue [hunk 7]
```diff
@@ -126,15 +126,15 @@
...
 
-        const type = await this.localStore.graknTx.getSchemaConcept(this.currentType);
+        const type = await graknTx.getSchemaConcept(this.currentType);
 
         this.nodeAttributes = await Promise.all((await (await type.attributes()).collect()).map(type => type.label()));
         this.nodeAttributes.sort();
         this.currentTypeSavedAttributes = NodeSettings.getTypeLabels(this.currentType);
 
-        await this.localStore.closeGraknTx();
+        await graknTx.close();
```
Comment (marco-scoppetta) [inline workbase/src/renderer/components/Visualiser/RightBar/SettingsTab/DisplaySettingsPanel.vue]: no need to await on close, we don't have to perform any specific operation

File: workbase/src/renderer/components/Visualiser/VisualiserStore.js [hunk 8]
```diff
@@ -87,7 +87,7 @@ const actions = {
         .sort());
 
     this.metaTypeInstances = metaTypeInstances;
-    await this.closeGraknTx();
+    await graknTx.close();
```
Comment (marco-scoppetta) [inline workbase/src/renderer/components/Visualiser/VisualiserStore.js]: no need to wait

File: workbase/src/renderer/components/Visualiser/VisualiserStore.js [hunk 9]
```diff
@@ -145,16 +145,21 @@ const methods = {
...
       this.updateCanvasData();
+
+      data.nodes = await VisualiserUtils.computeAttributes(data.nodes);
+
+      this.visFacade.updateNode(data.nodes);
+
       this.loadingQuery = false;
 
-      await this.closeGraknTx();
+      await graknTx.close();
```
Comment (marco-scoppetta) [inline workbase/src/renderer/components/Visualiser/VisualiserStore.js]: 🙅‍♂️

**Pull request merged**